src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -553,8 +553,8 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
             auto &task_queue = scheduling_key_entry.task_queue;
             while (!task_queue.empty()) {
               auto &task_spec = task_queue.front();
-              RAY_UNUSED(task_finisher_->MarkTaskReturnObjectsFailed(
-                  task_spec, rpc::ErrorType::RUNTIME_ENV_SETUP_FAILED));
+              RAY_UNUSED(task_finisher_->FailOrRetryPendingTask(
```

Comment:
This change makes the code harder to read. It was clear it would never get retried before, could we preserve that?

Reply:
I think the previous code was wrong: `MarkTaskReturnObjectsFailed` doesn't remove the task from `submissible_tasks_` in task_manager.cc, so we have memory leak in this case.

But yea, I can have a `FailPendingTask` method to make it clear that it never gets retried.